### ISSUE TITLE
Missing ROR IDs -- have they been deleted?

### ISSUE THREAD

**Issue by paulmillar** [ror-community/ror-api#199]
Hi,

Using the script `scan-wikidata-for-orphans` (available from [this repo](https://github.com/paulmillar/PIC-to-ROR/)), I discovered some 33 organisations in Wikidata for which Wikidata records a ROR identifier and that ROR identifier is missing from the latest ROR data dump from Zenodo.

When I checked some of these URLs manually, by putting the URL (e.g., https://ror.org/02n7qrg46) in my browser,  I see only a blank page.  This is the same behaviour as when I use an ID that does not exist; e.g., https://ror.org/0-no-such-id.

[The script's output](https://github.com/ror-community/ror-api/files/7726923/wikidata-orphans.txt) lists all the organisations with this problem, but I've included a few examples directly here:

- [Q1235608](http://www.wikidata.org/entity/Q1235608) has an orphaned link to [02n7qrg46](https://ror.org/02n7qrg46).
- [Q600624](http://www.wikidata.org/entity/Q600624) has an orphaned link to [01s9db082](https://ror.org/01s9db082).
- [Q3778182](http://www.wikidata.org/entity/Q3778182) has an orphaned link to [012qxnm65](https://ror.org/012qxnm65).
- [Q4533707](http://www.wikidata.org/entity/Q4533707) has an orphaned link to [00gayfz68](https://ror.org/00gayfz68).
- ... etc ...

I'm curious how this problem came about.

I don't imagine that someone added a random ROR ID to Wikidata.

I thought that, perhaps, someone made a mistake when entering the data.  To test this, I took the first example (Q1235608, "Paris Diderot University") and tried to find it in ROR.  It does not exist.  Wikidata claims that the university was renamed in 2019 to Q55849612 ("University of Paris").  Interestingly, Q55849612 links to the ROR ID 05f82e368, which _does_ exist in the ROR data dump.

Therefore, in this specific case, it looks like a new ROR entry was created when the university changed its name and the old entry was deleted.

The cause for the other three examples (listed above) is not so clear.  Where these ROR entries deleted deliberately?  If so, why?

Personally, I am somewhat surprised that ROR would intentionally delete an ID: I had expected ROR IDs to be a persistent identifier.

In a quick search, I couldn't find any policy about deleting ROR identifiers.  If this is done intentionally then it would be better to include some way of discovering that identifiers have been deleted.  Issue #99 mentions tombstone pages (which is one possible solution), but unfortunately that issue's description is (currently) too vague to know what it is addressing.

Cheers,
Paul.

**Comment by mariagould** [comment]
Hi @paulmillar ROR does not intentionally delete records and will never delete IDs. Until recently, all ROR registry updates were synced to GRID. When GRID merged, redirected, or deprecated a record, the same thing would reflect in ROR. However, ROR does not yet have the capability to redirect IDs or present a tombstone for IDs that have been deprecated. This is something that we will be working on in the new year (e.g., as per #99) as part of an effort to have comprehensive practices and policies in place for handling IDs for organization records that have changed status.

**Comment by Jason-Gush** [comment]
Great to see this being worked on as there's some urgency added to the issue by GRID's recent closing.  
In migrating we discovered that we have records for 14 Organisation with GRIDs that are not found in 2021-09-23-ror-grid-equivalents.zip as they have been closed, merged or otherwise deprecated.  E.g.,  grid.7452.4 Paris Diderot University and grid.7452.4 Université Paris Diderot - Paris 7, as mentioned they've both been merged into grid.508487.6 Université Paris Cité.  In the past (and my bad for not sorting this while the GRID registry was still live) my users would have hit https://grid.ac/grid.7452.4 and been redirected to https://grid.ac/grid.508487.6 which has a valid ROR (https://ror.org/05f82e368).  This history is available in the last grid.json dump, i.e., {"id": "grid.7452.4", "status": "redirected", "redirect": "grid.508487.6"} and it would be nice to see ROR behaving similarly.

**Comment by mariagould** [comment]
@Jason-Gush thanks for your input! We definitely understand that ROR needs to address this. In case you haven't seen it yet, we have put together a draft proposal for how ROR should handle organizations that become inactive, and we are inviting community feedback through July 15: https://bit.ly/ror-inactive-orgs.

**Comment by lizkrznarich** [comment]
Hi @Jason-Gush we actually have a draft proposal about handling inactive orgs in ROR at the moment. Resulting approach/metadata structure will also be applied to orgs that "fell out" of ROR between GRID releases. Would appreciate any thoughts you might have! https://bit.ly/ror-inactive-orgs

**Comment by paulmillar** [comment]
At least for me, the earlier explanation wasn't particularly illuminating.  As a comment in the above public proposal, @lizkrznarich provided what (at least, to me) was a much clearer explanation on how this happened.  I've copied her comment into this issue:

> This is because when ROR previously synced with GRID, ROR only included records with status = active per [this issue]. A follow on project to [handle inactive organizations in ROR] will be to restore all of the missing records, along with appropriate status information. Under the current curation process (which is no longer interlined with GRID), records are not removed.

If I've understood correctly (and expanding Liz's explanation somewhat), the problem occurred during the period where ROR were synchronised against GRID.  At some point, GRID updated the status of the affected organisations.  This change happened after ROR had assigned the affected organisations ROR IDs.  When ROR next synchronised against GRID, these organisations no longer had status=active and the ROR record was "lost".

This provides (at least, for me) a clear explanation on how ROR IDs have disappeared over time.

**Comment by paulmillar** [comment]
By chance, I came across a dataset [doi:10.5281/zenodo.6863846](https://doi.org/10.5281/zenodo.6863846) "ROR Identifiers That Have Disappeared" from [Ted Habermann](https://orcid.org/0000-0003-3585-6733).

I haven't checked the veracity of the dataset, but it seems to detail problems with the same underlying problem.

**Comment by lizkrznarich** [comment]
Thanks @paulmillar, Ted is an active ROR community member, so we're aware of his work on this and other topics. Internally, we have also identified the missing records by diff-ing all the data dumps from Nov 2018 to Sep 2021. We've created stub records for these and the curation team is now working to fill in information that is missing/incorrect/no longer schema valid. I'm also starting the [work to support additional status information in records](https://github.com/orgs/ror-community/projects/3#column-4886722) this week, and once that is complete, we will add the missing records back to the registry.

**Comment by amandafrench** [comment]
@paulmillar After a lot of work by @lizkrznarich and @adambuttrick and indeed the whole ROR community last year, this issue was resolved in December 2022: https://ror.readme.io/changelog/2022-12-01-organization-status-changes The example URLs you give above now resolve, usually to inactive records. We're all glad to have that sorted!
